weather.py

```diff
@@ -90,7 +163,35 @@ def generate_summary(weather_data):
     Returns:
         A string containing the summary information.
     """
-    pass
+
+    #establish variables & creating new lists by looping
+    numberOfDays = str(len(weather_data))
+    summary = ""
+
+    # Generating new lists for dates, min & max from weather_date list indexes
+    datesList = [date[0] for date in weather_data]
+    minTempList = [min[1] for min in weather_data]  #https://www.delftstack.com/howto/python/one-line-for-loop-python/
+    maxTempList = [max[2] for max in weather_data]
+
+
+    minTemp =  format_temperature(convert_f_to_c(find_min(minTempList)[0]))
+    maxTemp = format_temperature(convert_f_to_c(find_max(maxTempList)[0]))
+    minDate = convert_date(datesList[find_min(minTempList)[1]])
+    maxDate = convert_date(datesList[find_max(maxTempList)[1]])
```

Comment:
Minor suggestion: You can also use tuple unpacking here to avoid needing to reference index [0] or [1], since `find_min` returns a tuple, e.g.
```suggestion
    min_temp_f, min_temp_index = find_min(minTempList)
    minTemp =  format_temperature(convert_f_to_c(min_temp_f))
    minDate = convert_date(datesList[min_temp_index])
```